kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/ElementHandler.kt

```diff
@@ -129,6 +130,21 @@ interface ElementHandler {
    */
   fun methodAnnotations(classJvmName: String, methodSignature: JvmMethodSignature): List<AnnotationSpec>
 
+  /**
+   * Looks up a given class method given its [JvmMethodSignature] and returns any thrown types
+   * found on it. Used for [Throws][@Throws]
+   *
+   * @param classJvmName The JVM name of the class (example: `"org/foo/bar/Baz$Nested"`).
+   * @param methodSignature The method with annotations to look up.
+   * @param isConstructor Indicates if [methodSignature] is a constructor.
+   * @return the set of found thrown types, or null.
```

Comment:
nit: Any upside in returning null vs an empty set?

Reply:
null allows us to lean on kotlin nullability rather than checking if the returned annotations are empty

Reply:
Collections should almost never be nullable unless there's some semantic meaning between non-null vs. null and empty vs. non-empty. For things like an allow/disallow list, null might mean allow all whereas empty list means allow nothing.

Since you don't need the ternary state, empty list makes more sense here. You already always have the problem of dealing with empty lists in collections. Adding null just complicates things.

Reply:
fair enough, will change 👍